content-src/asrouter/components/ModalOverlay/ModalOverlay.jsx

```diff
@@ -1,28 +1,45 @@
 import React from "react";
 
-export const ModalOverlayWrapper = props => (<>
-  <div className={`modalOverlayOuter ${props.active ? "active" : ""}`} />
-  <div className={`modalOverlayInner ${props.active ? "active" : ""} ${props.innerClassName || ""}`}>
-    {props.children}
-  </div>
-</>);
+export class ModalOverlayWrapper extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  onKeyDown(event) {
+    if (event.key === "Escape") {
+      this.props.onClose();
+    }
+  }
 
-export class ModalOverlay extends React.PureComponent {
   componentWillMount() {
-    this.setState({active: true});
-    document.body.classList.add("modal-open");
+    this.props.document.addEventListener("keydown", this.onKeyDown);
+    this.props.document.body.classList.add("modal-open");
   }
 
   componentWillUnmount() {
-    document.body.classList.remove("modal-open");
-    this.setState({active: false});
+    this.props.document.removeEventListener("keydown", this.onKeyDown);
+    this.props.document.body.classList.remove("modal-open");
+  }
+
+  render() {
+    const {props} = this;
+    return (<React.Fragment>
+      <div className={`modalOverlayOuter active`} onClick={props.onClose} />
```

Comment:
Setting `role="presentation"` on this `<div onClick…>` makes these a11y errors go away:
```
  28:7  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  28:7  error  Static HTML elements with event handlers require a role                                         jsx-a11y/no-static-element-interactions
```
As I guess it no longer considers it "visible"